roles/openshift_docker_facts/tasks/main.yml

```diff
@@ -37,3 +37,43 @@
 - set_fact:
     docker_options: "{{ openshift.docker.options | default(omit) }}"
   when: not openshift.docker.hosted_registry_insecure | default(False) | bool
+
+# Avoid docker 1.9 when installing origin < 1.2 or OSE < 3.2 on RHEL/Centos and
+# See: https://bugzilla.redhat.com/show_bug.cgi?id=1304038
+- name: Set docker defaults
+  set_fact:
+    docker_version: ''
+    docker_downgrade: False
+    repoquery_cmd: "{{ 'dnf repoquery --latest-limit 1 -d 0' if ansible_pkg_mgr == 'dnf' else 'repoquery' }}"
+
+- name: Gather common package version
+  command: >
+    {{ repoquery_cmd }} --qf '%{version}' "{{ openshift.common.service_type}}"
+  register: common_version
+  failed_when: false
+  changed_when: false
+  when: not openshift.common.is_atomic | bool
+  
+- name: Gather installed version of docker
+  command: >
+    {{ repoquery_cmd }} --installed --qf '%{version}' docker
+  register: installed_docker_version
+  failed_when: false
+  changed_when: false
+  when: not openshift.common.is_atomic | bool
+  
+- name: Set docker version to be installed
+  set_fact:
+    docker_version: "{{ '1.8.2' }}"
+  when: " ( common_version.stdout | default('0.0') | version_compare('3.2','<') and openshift.common.service_type == 'atomic-openshift' ) or
+          ( common_version.stdout | default('0.0') | version_compare('1.2','<') and openshift.common.service_type == 'origin' )"
+
+# for some reason couldn't use installed_docker_version.stdout directly below
+- set_fact:
+    _installed: installed_docker_version.stdout | default('0.0')
```

Comment:
I'd move this up before 'Set docker version to be installed' and add another fact for:
```
l_available: common_version.stdout | default('0.0')
```
Also, change `_installed` to `l_installed` to avoid breaking ansible 2.0